src/index.js

```diff
@@ -151,6 +168,64 @@ function getAcronym(string) {
   return acronym
 }
 
+/**
+ * Returns a score base on the case of the testString
+ * @param {String} testString - the string to test against
+ * @returns {Number} the number of the ranking,
+ * based on the case between 0 and 1 for how the testString matches the case
+ */
+function getCaseRanking(testString) {
+  const containsUpperCase = testString.toLowerCase() !== testString
+  const containsDash = testString.includes('-')
+  const containsUnderscore = testString.includes('_')
+
+  if (!containsUpperCase && !containsUnderscore && containsDash) {
+    return caseRankings.KEBAB
+  }
+
+  if (!containsUpperCase && containsUnderscore && !containsDash) {
+    return caseRankings.SNAKE
+  }
+
+  if (containsUpperCase && !containsDash && !containsUnderscore) {
+    const startsWithUpperCase = testString[0].toUpperCase() === testString[0]
+    if (startsWithUpperCase) {
+      return caseRankings.PASCAL
+    }
+
+    return caseRankings.CAMEL
+  }
+
+  return caseRankings.NO_CASE
+}
+
+/**
+ * Returns whether the stringToRank is one of the case parts in the testString
+ * @param {String} testString - the string to test against
+ * @param {String} stringToRank - the string to rank
+ * @param {Number} caseRanking - the ranking score based on case of testString
+ * @returns {Boolean} whether the stringToRank is one of the case parts in the testString
+ */
+function isPartialOfCase(testString, stringToRank, caseRanking) {
```

Comment:
I'm not sure I understand this function. Could you explain what its purpose is and how it handles different inputs?

Reply:
Example: It gets a teststring like `helloWorld`. This has then a ranking of `caseRankings.CAMEL` (from the`caseRanking` function). Then when you search for `world` it gets the position of the stringToRank and then uses that to determine whether it is a part of `helloWorld` or `hello_world`. With part I mean `hello` or `world`. This means when you search `world` it returns true, but when you search `ello` or `oworld` it returns false. This is the same for the other cases like snake and kebab.

The nice thing is that you don't have to worry about the first character, since that checked earlier with `STARTS_WITH`. For kebab and snake I look at the character before to check if it's a dash or underscore and for the camel and pascal I check if the letter itsself is uppercase.

Reply:
Ok, I think that makes sense. Thanks for adding the examples :+1: